### PR TITLE
aws: util: use memcpy instead of strncat and set null byte

### PR DESCRIPTION
**Pull request by PettitWesley**
Signed-off-by: Wesley Pettit <wppttt@amazon.com>

----

Fluent Bit is licensed under Apache 2.0, by submitting this pull request I understand that this code will be released under the terms of that license.

### CODE REVIEW

File: src/aws/flb_aws_util.c [hunk 1]
```diff
@@ -56,7 +56,7 @@ char *flb_aws_endpoint(char* service, char* region)
 
     len += strlen(service);
     len += strlen(region);
-    len ++; /* null byte */
+    len++; /* null byte */
 
     endpoint = flb_malloc(sizeof(char) * len);
```
Comment (edsiper) [inline src/aws/flb_aws_util.c:61]: sizeof(char) always equals 1 , so it's safe to use only ```len```

File: src/aws/flb_aws_util.c [hunk 2]
```diff
@@ -67,7 +67,8 @@ char *flb_aws_endpoint(char* service, char* region)
     snprintf(endpoint, len, AWS_SERVICE_ENDPOINT_FORMAT, service, region);
 
     if (is_cn) {
-        strncat(endpoint, ".cn", 3);
+        memcpy(endpoint + (len - 4), ".cn", 3);
```
Comment (edsiper) [inline src/aws/flb_aws_util.c:70]: on this case, if you want to write "after" the last valid byte of ```endpoint```, I suggest you check the return value of snprintf() which returns the number of bytes written to endpoint, then you can do something like:

```c
bytes = snprintf(endpoint...)

memcpy(endpoint + bytes, ".cn", 3)
len = bytes + 3;
endpoint[len] = '\0'
```

**Comment by PettitWesley** [comment]
@edsiper updated

**Comment by edsiper** [comment]
thanks

**Pull request merged**